pkg/controller/upload-controller.go

```diff
@@ -486,10 +486,14 @@ func addUploadControllerWatches(mgr manager.Manager, importController controller
 	return nil
 }
 
+// getScratchPvcName returns the name given to scratch pvc
+func getScratchPvcName(name string) string {
+	return util.GetPvcNameWithSuffix(name, "scratch")
+}
+
 // getUploadResourceName returns the name given to upload resources
 func getUploadResourceName(name string) string {
-	// TODO revisit naming, could overflow
-	return "cdi-upload-" + name
+	return util.GetPodNameWithPrefix(name, "cdi-upload")
```

Comment:
In general, I'd think the "base" part of the name would be the const part (in this case `cdi-upload`) and would go first.

Specifically for upload, I think we should try to have the naming be compatible (there may be an upload pod running during an upgrade) or store the pod name in an annotation of the PVC once calculated.  If the annotation doesn't exist, use the old scheme to determine pod name.

Reply:
Here `GetPodNameWithPrefix` takes name and prefix. Prefix goes first and is not changed (unless it is too long then it is dropped). If name is too long it is shortened. Do you think the order of arguments should be changed here to improve readability?

The thing with backwards compatibility is really important I need to check this. But if the name is not too long then current method returns exactly the same results as previous ("cdi-upload-${name}"). I wonder what happened in previous version if name was too long. That might be not compatible.

Reply:
I have done it as you suggest, so there is now need to write our own namer(I have used openshift/library-go) , and also naming is backwards compatible.